### PR TITLE
mg: add new package - microscopic emacs style editor

### PR DESCRIPTION
**Pull request by nxhack**
Maintainer: me 
Compile tested: head r10012-5c742c8, aarch64_cortex-a53_gcc-7.4.0_musl
Run tested: aarch64 (ec2 a1 docker)

Description:
New package for Emacsian.
You can use "Dired", which is Emacsian's favorite.

Signed-off-by: Hirokazu MORIKAWA <morikw2@gmail.com>

### CODE REVIEW

File: utils/mg/patches/900-musl_compatibility.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
+--- a/fileio.c
++++ b/fileio.c
+@@ -23,6 +23,13 @@
+ #include <string.h>
+ #include <unistd.h>
+ 
++#ifndef DEFFILEMODE
++#define DEFFILEMODE (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)
++#endif
++#ifndef MAXNAMLEN
```
Comment (neheb) [inline utils/mg/patches/900-musl_compatibility.patch]: A quick check shows musl defining this in sys/param.h and dirent.h for uclibc-ng.

Comment (neheb) [reply utils/mg/patches/900-musl_compatibility.patch]: A closer look shows MAXNAMLEN to be a BSD define whereas the POSIX one is NAME_MAX.

Comment (nxhack) [reply utils/mg/patches/900-musl_compatibility.patch]: musl : DEFFILEMODE is undefined, MAXNAMLEN is define in sys/param.h
uclibc:  DEFFILEMODE is definedin  sys/stat.h, MAXNAMLEN is defined in dirent.h

Anyway, my patch works well in both environments.

Comment (neheb) [reply utils/mg/patches/900-musl_compatibility.patch]: Hmm? I git grepped the codebase and found only one usage of MAXNAMLEN in some DragonflyBSD header. It seems MAXNAMLEN was already replaced by NAME_MAX in fileio.c.

Comment (neheb) [reply utils/mg/patches/900-musl_compatibility.patch]: Yeah. https://github.com/ibara/mg/commit/bc00df55d725b3a1e843003afb17d24dbaf4d23f

Comment (nxhack) [reply utils/mg/patches/900-musl_compatibility.patch]: Oh. I did not know that MAXNAMLEN was not used in the new version.

**Pull request merged**